src/services/api.service.ts

```diff
@@ -29,6 +33,9 @@ export const transformMediaItem = (item: PlaylistItem, playlist?: Playlist) => {
     backgroundImage: generateImageData(config, ImageProperty.BACKGROUND, item),
     channelLogoImage: generateImageData(config, ImageProperty.CHANNEL_LOGO, item),
     mediaOffers: item.productIds ? filterMediaOffers(offerKeys, item.productIds) : undefined,
+    scheduledStart: item['VCH.ScheduledStart'] ? parseISO(item['VCH.ScheduledStart'] as string) : undefined,
+    scheduledEnd: item['VCH.ScheduledEnd'] ? parseISO(item['VCH.ScheduledEnd'] as string) : undefined,
+    mediaStatus: getMediaStatusFromEventState(item),
   };
```

Comment:
The `getMediaStatusFromEventState` utility makes use of the transformed `scheduledStart` / `scheduledEnd` properties which are not present yet in `item`.

Should we skip precalculating the `mediaStatus` here and use the `useLiveEvent` hook and `getMediaStatusFromEventState` util instead so we always get the most current media status?

Alternatively, we can split the transformed item into two steps:

```ts
  const tempItem = {
    ...item,
    backgroundImage: generateImageData(config, ImageProperty.BACKGROUND, item),
    channelLogoImage: generateImageData(config, ImageProperty.CHANNEL_LOGO, item),
    mediaOffers: item.productIds ? filterMediaOffers(offerKeys, item.productIds) : undefined,
    scheduledStart: item['VCH.ScheduledStart'] ? parseISO(item['VCH.ScheduledStart'] as string) : undefined,
    scheduledEnd: item['VCH.ScheduledEnd'] ? parseISO(item['VCH.ScheduledEnd'] as string) : undefined,
  };

  // mediaStatus uses the `scheduledStart` and `scheduledEnd` properties
  return {
    ...tempItem,
    mediaStatus: getMediaStatusFromEventState(tempItem),
  };
```

Reply:
I need the mediaStatus in the shelf's so I went for the second solution